app/controllers/CyaMaintainProfileController.scala

```diff
@@ -74,6 +74,34 @@ class CyaMaintainProfileController @Inject() (
       case Left(errors) =>
         Future.successful(logErrorsAndContinue(errorMessage, continueUrl, errors))
     }
+  }
+
+  def onPageLoadNiphl(): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
+    TraderProfile.validateHasNiphl(request.userAnswers) match {
+      case Right(_)     =>
+        val list = SummaryListViewModel(
+          rows = Seq(
+            HasNiphlSummary.rowUpdate(request.userAnswers)
+          ).flatten
+        )
+        Ok(view(list, routes.CyaMaintainProfileController.onSubmitNiphl))
+      case Left(errors) =>
+        logErrorsAndContinue(errorMessage, continueUrl, errors)
```

Comment:
the continue url doesn't work on the journey recovery page - check https://github.com/hmrc/trader-goods-profiles-frontend/pull/412 for a fix